src/vs/workbench/contrib/notebook/browser/services/notebookKernelServiceImpl.ts

```diff
@@ -245,10 +245,11 @@ export class NotebookKernelService extends Disposable implements INotebookKernel
 		const selectedId = this._notebookBindings.get(NotebookTextModelLikeId.str(notebook));
 		const selected = selectedId ? this._kernels.get(selectedId)?.kernel : undefined;
 		const suggestions = kernels.filter(item => item.instanceAffinity > 1 && item.kernel !== selected).map(item => item.kernel);
+		const hidden = kernels.filter(item => item.instanceAffinity < 0).map(item => item.kernel);
 		if (!suggestions.length && all.length) {
 			suggestions.push(all[0]);
 		}
-		return { all, selected, suggestions };
+		return { all, selected, suggestions, hidden };
```

Comment:
Instead of filtering all at this point it seemed to match the current usage to keep all as all, then just provide hidden to indicate the kernels marked as hidden.